geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/index/s3/S3IndexKeySpace.scala

```diff
@@ -293,30 +291,20 @@ class S3IndexKeySpace(val sft: SimpleFeatureType,
   }
 
   /**
-    * Determines if the ranges generated by `getRanges` are sufficient to fulfill the query,
-    * or if additional filtering needs to be done
-    *
-    * @param config data store config
-    * @param values index values @see getIndexValues
-    * @param hints  query hints
-    * @return
-    */
-  override def useFullFilter(values: Option[S3IndexValues],
-                             config: Option[GeoMesaDataStoreFactory.GeoMesaDataStoreConfig],
-                             hints: Hints): Boolean = {
-
-    // if the user has requested strict bounding boxes, we apply the full filter
-    // if we have a complicated geometry predicate, we need to pass it through to be evaluated
-    // if we have unbounded dates, we need to pass them through as we don't have z-values for all periods
-
-    // if the spatial predicate is rectangular (e.g. a bbox), the index is fine enough that we
-    // don't need to apply the filter on top of it. this may cause some minor errors at extremely
-    // fine resolutions, but the performance is worth it
-    val looseBBox = Option(hints.get(LOOSE_BBOX)).map(Boolean.unbox).getOrElse(config.forall(_.looseBBox))
-    def unboundedDates: Boolean = values.exists(_.temporalUnbounded.nonEmpty)
-    def complexGeoms: Boolean = values.exists(_.geometries.values.exists(g => !GeometryUtils.isRectangular(g)))
-    !looseBBox || unboundedDates || complexGeoms
-  }
+   * Determines if the ranges generated by `getRanges` are sufficient to fulfill the query,
+   * or if additional filtering needs to be done
+   *
+   * Because we have a tiered date index, we always need to use the full filter
+   *
+   * @param config data store config
+   * @param values index values @see getIndexValues
+   * @param hints  query hints
+   * @return
+   */
+  override def useFullFilter(
+      values: Option[S3IndexValues],
+      config: Option[GeoMesaDataStoreFactory.GeoMesaDataStoreConfig],
+      hints: Hints): Boolean = true
```

Comment:
Is there an error in the previous logic?

Reply:
well, due to the index structure being a tiered `<s2><time>`, the ranges aren't precise unless you specify every single s2 cell